<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ColorDialog</class>
 <widget class="QDialog" name="ColorDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>303</width>
    <height>291</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>303</width>
    <height>291</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>303</width>
    <height>291</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Select Color</string>
  </property>
  <property name="windowIcon">
   <iconset resource="pattern.qrc">
    <normaloff>:/Resources/ColorPicker.png</normaloff>:/Resources/ColorPicker.png</iconset>
  </property>
  <layout class="QGridLayout" name="gridLayout_6">
   <property name="leftMargin">
    <number>3</number>
   </property>
   <property name="topMargin">
    <number>3</number>
   </property>
   <property name="rightMargin">
    <number>3</number>
   </property>
   <property name="bottomMargin">
    <number>3</number>
   </property>
   <property name="spacing">
    <number>3</number>
   </property>
   <item row="5" column="3">
    <widget class="QPushButton" name="cancel">
     <property name="text">
      <string>Cancel</string>
     </property>
    </widget>
   </item>
   <item row="0" column="3" rowspan="3">
    <layout class="QGridLayout" name="gridLayout_4">
     <property name="spacing">
      <number>3</number>
     </property>
     <item row="0" column="0" colspan="2">
      <widget class="QListWidget" name="listWidget_pallet">
       <property name="movement">
        <enum>QListView::Static</enum>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QPushButton" name="button_add">
       <property name="maximumSize">
        <size>
         <width>55</width>
         <height>23</height>
        </size>
       </property>
       <property name="text">
        <string>+</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QPushButton" name="button_remove">
       <property name="maximumSize">
        <size>
         <width>55</width>
         <height>23</height>
        </size>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="0" column="0" colspan="3">
    <layout class="QGridLayout" name="gridLayout_5">
     <property name="spacing">
      <number>3</number>
     </property>
     <item row="1" column="1">
      <widget class="QPushButton" name="button_pick">
       <property name="maximumSize">
        <size>
         <width>24</width>
         <height>24</height>
        </size>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="pattern.qrc">
         <normaloff>:/Resources/ColorPicker.png</normaloff>:/Resources/ColorPicker.png</iconset>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="ColorPreview" name="preview"/>
     </item>
     <item row="0" column="0" rowspan="2">
      <widget class="ColorWheel" name="wheel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>150</width>
         <height>150</height>
        </size>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="1" column="1">
    <widget class="GradientSlider" name="slide_alpha">
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item row="1" column="2">
    <widget class="QSpinBox" name="spin_alpha">
     <property name="maximum">
      <number>255</number>
     </property>
    </widget>
   </item>
   <item row="2" column="0" rowspan="4" colspan="3">
    <widget class="QTabWidget" name="tabWidget">
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>123</height>
      </size>
     </property>
     <property name="tabPosition">
      <enum>QTabWidget::South</enum>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>HSV</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_2">
       <property name="leftMargin">
        <number>3</number>
       </property>
       <property name="topMargin">
        <number>3</number>
       </property>
       <property name="rightMargin">
        <number>3</number>
       </property>
       <property name="bottomMargin">
        <number>3</number>
       </property>
       <property name="spacing">
        <number>3</number>
       </property>
       <item row="0" column="0">
        <widget class="QLabel" name="label_6">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>H:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="GradientSlider" name="slide_hue">
         <property name="maximum">
          <number>359</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QSpinBox" name="spin_hue">
         <property name="wrapping">
          <bool>true</bool>
         </property>
         <property name="maximum">
          <number>359</number>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_7">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>S:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="GradientSlider" name="slide_saturation">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QSpinBox" name="spin_saturation">
         <property name="maximum">
          <number>255</number>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_5">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>V:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="GradientSlider" name="slide_value">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QSpinBox" name="spin_value">
         <property name="maximum">
          <number>255</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>RGB</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_3">
       <property name="leftMargin">
        <number>3</number>
       </property>
       <property name="topMargin">
        <number>3</number>
       </property>
       <property name="rightMargin">
        <number>3</number>
       </property>
       <property name="bottomMargin">
        <number>3</number>
       </property>
       <property name="spacing">
        <number>3</number>
       </property>
       <item row="0" column="0">
        <widget class="QLabel" name="label">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>R:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="GradientSlider" name="slide_red">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QSpinBox" name="spin_red">
         <property name="maximum">
          <number>255</number>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_2">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>G:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="GradientSlider" name="slide_green">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QSpinBox" name="spin_green">
         <property name="maximum">
          <number>255</number>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_3">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>B:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="GradientSlider" name="slide_blue">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QSpinBox" name="spin_blue">
         <property name="maximum">
          <number>255</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>Hex</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout">
       <property name="leftMargin">
        <number>3</number>
       </property>
       <property name="topMargin">
        <number>3</number>
       </property>
       <property name="rightMargin">
        <number>3</number>
       </property>
       <property name="bottomMargin">
        <number>3</number>
       </property>
       <property name="spacing">
        <number>3</number>
       </property>
       <item row="0" column="0">
        <widget class="QLabel" name="label_8">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Hex:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="edit_hex">
         <property name="font">
          <font>
           <family>Monospace</family>
          </font>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>51</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="label_4">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>A:</string>
     </property>
    </widget>
   </item>
   <item row="3" column="3">
    <widget class="Line" name="line">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item row="4" column="3">
    <widget class="QPushButton" name="ok">
     <property name="text">
      <string>Ok</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>ColorWheel</class>
   <extends>QWidget</extends>
   <header>ColorWheel.h</header>
  </customwidget>
  <customwidget>
   <class>GradientSlider</class>
   <extends>QSlider</extends>
   <header>GradientSlider.h</header>
  </customwidget>
  <customwidget>
   <class>ColorPreview</class>
   <extends>QWidget</extends>
   <header>ColorPreview.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="pattern.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>slide_hue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spin_hue</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>128</x>
     <y>209</y>
    </hint>
    <hint type="destinationlabel">
     <x>174</x>
     <y>208</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin_hue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>slide_hue</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>174</x>
     <y>208</y>
    </hint>
    <hint type="destinationlabel">
     <x>128</x>
     <y>209</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slide_saturation</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spin_saturation</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>128</x>
     <y>234</y>
    </hint>
    <hint type="destinationlabel">
     <x>174</x>
     <y>233</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin_saturation</sender>
   <signal>valueChanged(int)</signal>
   <receiver>slide_saturation</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>174</x>
     <y>233</y>
    </hint>
    <hint type="destinationlabel">
     <x>128</x>
     <y>234</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slide_value</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spin_value</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>128</x>
     <y>259</y>
    </hint>
    <hint type="destinationlabel">
     <x>174</x>
     <y>258</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin_value</sender>
   <signal>valueChanged(int)</signal>
   <receiver>slide_value</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>174</x>
     <y>258</y>
    </hint>
    <hint type="destinationlabel">
     <x>128</x>
     <y>259</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slide_red</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spin_red</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>55</x>
     <y>193</y>
    </hint>
    <hint type="destinationlabel">
     <x>174</x>
     <y>194</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin_red</sender>
   <signal>valueChanged(int)</signal>
   <receiver>slide_red</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>174</x>
     <y>194</y>
    </hint>
    <hint type="destinationlabel">
     <x>55</x>
     <y>193</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slide_green</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spin_green</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>55</x>
     <y>218</y>
    </hint>
    <hint type="destinationlabel">
     <x>174</x>
     <y>219</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin_green</sender>
   <signal>valueChanged(int)</signal>
   <receiver>slide_green</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>174</x>
     <y>219</y>
    </hint>
    <hint type="destinationlabel">
     <x>55</x>
     <y>218</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slide_alpha</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spin_alpha</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>135</x>
     <y>179</y>
    </hint>
    <hint type="destinationlabel">
     <x>181</x>
     <y>178</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin_alpha</sender>
   <signal>valueChanged(int)</signal>
   <receiver>slide_alpha</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>181</x>
     <y>178</y>
    </hint>
    <hint type="destinationlabel">
     <x>135</x>
     <y>179</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slide_blue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spin_blue</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>55</x>
     <y>243</y>
    </hint>
    <hint type="destinationlabel">
     <x>174</x>
     <y>244</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin_blue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>slide_blue</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>174</x>
     <y>244</y>
    </hint>
    <hint type="destinationlabel">
     <x>55</x>
     <y>243</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cancel</sender>
   <signal>clicked()</signal>
   <receiver>ColorDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>299</x>
     <y>287</y>
    </hint>
    <hint type="destinationlabel">
     <x>197</x>
     <y>322</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ok</sender>
   <signal>clicked()</signal>
   <receiver>ColorDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>299</x>
     <y>261</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>292</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>wheel</sender>
   <signal>signalColorSelected(QColor)</signal>
   <receiver>ColorDialog</receiver>
   <slot>slotUpdateWidgets()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>141</x>
     <y>39</y>
    </hint>
    <hint type="destinationlabel">
     <x>144</x>
     <y>-6</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slide_alpha</sender>
   <signal>valueChanged(int)</signal>
   <receiver>ColorDialog</receiver>
   <slot>slotUpdateWidgets()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>21</x>
     <y>166</y>
    </hint>
    <hint type="destinationlabel">
     <x>47</x>
     <y>-13</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slide_red</sender>
   <signal>valueChanged(int)</signal>
   <receiver>ColorDialog</receiver>
   <slot>slotSetRgb()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>103</x>
     <y>203</y>
    </hint>
    <hint type="destinationlabel">
     <x>183</x>
     <y>47</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slide_green</sender>
   <signal>valueChanged(int)</signal>
   <receiver>ColorDialog</receiver>
   <slot>slotSetRgb()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>91</x>
     <y>229</y>
    </hint>
    <hint type="destinationlabel">
     <x>184</x>
     <y>31</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slide_blue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>ColorDialog</receiver>
   <slot>slotSetRgb()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>61</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>182</x>
     <y>17</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slide_hue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>ColorDialog</receiver>
   <slot>slotSetHsv()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>75</x>
     <y>198</y>
    </hint>
    <hint type="destinationlabel">
     <x>151</x>
     <y>145</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slide_saturation</sender>
   <signal>valueChanged(int)</signal>
   <receiver>ColorDialog</receiver>
   <slot>slotSetHsv()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>75</x>
     <y>223</y>
    </hint>
    <hint type="destinationlabel">
     <x>151</x>
     <y>145</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slide_value</sender>
   <signal>valueChanged(int)</signal>
   <receiver>ColorDialog</receiver>
   <slot>slotSetHsv()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>75</x>
     <y>248</y>
    </hint>
    <hint type="destinationlabel">
     <x>151</x>
     <y>145</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>slotSetRgb()</slot>
  <slot>slotSetHsv()</slot>
  <slot>slotSetColor(QColor)</slot>
  <slot>slotUpdateWidgets()</slot>
 </slots>
</ui>
